File: 3/python/a.py
with open("input.txt", "r") as f:
    wires_points = [line.split(",") for line in f.read().splitlines()]

def directions_to_points(directions):
    curr_point = [0, 0]
    points = []
    for direction in directions:
        if direction[0] == "R":
            for i in range(0, int(direction[1:])):
                curr_point = [curr_point[0]+1, curr_point[1]]
                points.append(tuple(curr_point))
        elif direction[0] == "L":
            for i in range(0, int(direction[1:])):
                curr_point = [curr_point[0]-1, curr_point[1]]
                points.append(tuple(curr_point))
        elif direction[0] == "U":
            for i in range(0, int(direction[1:])):
                curr_point = [curr_point[0], curr_point[1]+1]
                points.append(tuple(curr_point))
        elif direction[0] == "D":
            for i in range(0, int(direction[1:])):
                curr_point = [curr_point[0], curr_point[1]-1]
                points.append(tuple(curr_point))                                                                
    return points

def dist_calc(point_1, point_2):
    return abs(point_1[0] - point_2[0]) + abs(point_1[1] - point_2[1])

wires = []
for wire_points in wires_points:
    wires.append(directions_to_points(wire_points))

#cast to set to perform intersection operator to find common points
#cast to list for subscribabiltiy
#first element is ignored
common_points = list(set(wires[0]).intersection(set(wires[1])))

best_point = (0,0)
#you can never be too carefull
best_point_dist = float("inf")
for point in common_points:
    dist = dist_calc(point, (0,0))
    if dist <= best_point_dist:
        best_point_dist = dist
        best_point = point
        
print(best_point)
print(best_point_dist)

File: 3/scala/a.scala
import io.Source.fromFile
import math.abs
import scala.collection.mutable

// extracts the single line of intcode and stores it as an array
// each intcode memory location is given it's own element
val file = fromFile("input.txt").getLines
val wire_1 = file.next().split(",")
val wire_2 = file.next().split(",")

var wire_1_points: mutable.Set[(Int, Int)] = mutable.Set()
var wire_2_points: mutable.Set[(Int, Int)] = mutable.Set()

var x: Int = 0
var y: Int = 0
var i: Int = 0
var direction: Char = '.'
var length: Int = 0

//prdouces an exahustive list for each wire of every point that the wire crosses through
for (command <- wire_1){
    direction = command(0)
    length = command.slice(1,100).toInt
    if (direction == 'R') {
        for (i <- (1 to length)){
            x += 1
            wire_1_points.add((x,y))
        }
    }
    else if (direction == 'L') {
        for (i <- (1 to length)){
            x -= 1
            wire_1_points.add((x,y))
        }
    }
    else if (direction == 'U') {
        for (i <- (1 to length)){
            y += 1
            wire_1_points.add((x,y))
        }
    }
    else if (direction == 'D') {
        for (i <- (1 to length)){
            y -= 1
            wire_1_points.add((x,y))
        }
    }
}

x = 0
y = 0
for (command <- wire_2){
    direction = command(0)
    length = command.slice(1,100).toInt
    if (direction == 'R') {
        for (i <- (1 to length)){
            x += 1
            wire_2_points.add((x,y))
        }
    }
    else if (direction == 'L') {
        for (i <- (1 to length)){
            x -= 1
            wire_2_points.add((x,y))
        }
    }
    else if (direction == 'U') {
        for (i <- (1 to length)){
            y += 1
            wire_2_points.add((x,y))
        }
    }
    else if (direction == 'D') {
        for (i <- (1 to length)){
            y -= 1
            wire_2_points.add((x,y))
        }
    }
}
//intersection of the two sets
val common_points = wire_1_points & wire_2_points

var best_point: (Int, Int) = (0,0)
//very high just to be on the safe side
var best_point_distance: Int = 2000000000
var distance: Int = 0
for (point <- common_points) {
    //._1 gets the first element in a tuple
    //._2 gets the second element in a tuple
    distance = abs(point._1) + abs(point._2)
    if (distance < best_point_distance) {
        best_point = point
        best_point_distance = distance
    }
}
println(best_point)
println(best_point_distance)